crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -1,40 +1,140 @@
+//! It is also possible to use [`World::spawn_empty`](crate::world::World::spawn_empty) or [`Commands::spawn_empty`](crate::system::Commands::spawn_empty), which are similar but do not add any components to the entity.
+//! In either case, the returned [`Entity`] id is used to further interact with the entity.
+//! Once an entity is created, you will need its [`Entity`] id to progress the entity through its life cycle.
+//! This can be done through [`World::entity_mut`](crate::world::World::entity_mut) and [`Commands::entity`](crate::system::Commands::entity).
+//! Even if you don't store the id, you can still find the entity you spawned by searching for it in a [`Query`].
+//!
+//! **Insert:** Once an entity has been created, additional [`Bundle`]s can be inserted onto the entity.
+//! There are lots of ways to do this and lots of ways to configure what to do when a component in the bundle is already present on the entity.
+//! Each entity can only have 0 or 1 values for a component.
+//! See [`EntityWorldMut::insert`](crate::world::EntityWorldMut::insert) and [`EntityCommands::insert`](crate::system::EntityCommands::insert) for a start on how to do this.
+//!
+//! **Remove:** Components on an entity can be removed as well.
+//! See [`EntityWorldMut::remove`](crate::world::EntityWorldMut::remove) and [`EntityCommands::remove`](crate::system::EntityCommands::remove) for a start on how to do this.
+//!
+//! **Despawn:** Despawn an entity when it is no longer needed.
+//! This destroys it and all its components.
+//! The entity is no longer reachable through the [`World`], [`Commands`], or [`Query`]s.
+//! Note that this means an [`Entity`] id may refer to an entity that has since been despawned!
+//! Not all [`Entity`] ids refer to active entities.
+//! If an [`Entity`] id is used when its entity no longer exists, an [`EntityDoesNotExistError`] is emitted.
+//! Any [`System`](crate::system) could despawn entities; even if you never share an entity's id, it could still be despawned unexpectedly.
+//! Handle these errors gracefully.
+//!
+//! In short:
+//!
+//! - Entities are spawned through methods like [`World::spawn`](crate::world::World::spawn), which return an [`Entity`] id for the new entity.
+//! - Once spawned, they can be accessed and modified through [`Query`]s and other apis.
+//! - You can get the [`Entity`] id of an entity through [`Query`]s, so loosing an [`Entity`] id is not a problem.
+//! - Entities can have components inserted and removed via [`World::entity_mut`](crate::world::World::entity_mut) and [`Commands::entity`](crate::system::Commands::entity).
+//! - Entities are eventually despawned, destroying the entity and causing its [`Entity`] id to no longer refer to an entity.
+//! - Not all [`Entity`] ids point to actual entities, which makes many entity methods fallible.
+//!
+//! # Entity Ids
+//!
+//! As mentioned entities each have an [`Entity`] id, which is used to interact with that entity.
```

Comment:
I'm assuming you mean the entity structure has two fields: an integer of some kind for the row and an integer of some kind for the generation. 

There is just a lot of word "entity" thrown around so it might useful to reference the actual struct with "entity struct" literally and then name its fields. I don't know if I'm being clear here but basically I'm saying it's hard to know what entity is which.